workspaces/multi-source-security-viewer/plugins/multi-source-security-viewer/src/api/github.ts

```diff
@@ -0,0 +1,171 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  GithubActionsApi,
+  GITHUB_ACTIONS_ANNOTATION,
+  BuildStatus,
+} from '@backstage-community/plugin-github-actions';
+import { MssvApi, MssvApiResponse } from './mssv';
+import { Entity } from '@backstage/catalog-model';
+import { createApiRef } from '@backstage/core-plugin-api';
+import { PipelineRunResult } from '../models/pipelineRunResult';
+import { RunStatus } from '../types/pipelinerun';
+
+// Apiref to map with client
+export const mssvGithubActionsApiRef = createApiRef<MssvApi>({
+  id: 'plugin.mssv-api-githubactions.service',
+});
+
+const mapStatus = (status: string): RunStatus => {
+  const runStatus = BuildStatus[status as keyof typeof BuildStatus];
+  switch (runStatus) {
+    case BuildStatus.success:
+      return RunStatus.Succeeded;
+    case BuildStatus.failure:
+      return RunStatus.Failed;
+    case BuildStatus.pending:
+      return RunStatus.Pending;
+    case BuildStatus.running:
+      return RunStatus.Running;
+    default:
+      return RunStatus.Unknown;
+  }
+};
+
+export class MssvGithubActionsClient implements MssvApi {
+  private readonly githubActionsApi: GithubActionsApi;
+
+  constructor(options: { githubActionsApi: GithubActionsApi }) {
+    this.githubActionsApi = options.githubActionsApi;
+  }
+
+  async getPipelineSummary(options: {
+    entity: Entity;
+    page: number;
+    pageSize: number;
+  }): Promise<MssvApiResponse> {
+    const { entity, page, pageSize } = options;
+    const [owner, repo] = (
+      entity?.metadata.annotations?.[GITHUB_ACTIONS_ANNOTATION] ?? '/'
+    ).split('/');
+
+    if (!owner || !repo) {
+      return Promise.reject('No repo/owner provided');
+    }
+
+    const project = await this.githubActionsApi.listWorkflowRuns({
+      owner,
+      repo,
+      page: page + 1, // non zero-based
+      pageSize,
+    });
+
+    const projectWithLogs = await Promise.all(
+      project.workflow_runs.map(async run => {
+        const jobsList = await this.githubActionsApi.listJobsForWorkflowRun({
+          owner,
+          repo,
+          id: run.id,
+        });
+
+        const logs = await Promise.all(
+          jobsList.jobs.map(async job => {
+            const jobLogs =
+              await this.githubActionsApi.downloadJobLogsForWorkflowRun({
+                owner,
+                repo,
+                runId: job.id,
+              });
+            return jobLogs;
+          }),
+        ).then(res => res.join(' ')); // return as one string
```

Comment:
Github logs may expire over a period of time, if that happens then API will fail with `410`. can we add a catch block to handle these known errors ? so that only those failed requests are discarded and the rest of the rows should be rendered


No rows are rendered now due a api failure.
![image](https://github.com/user-attachments/assets/a88cbf14-5bb8-49af-8bc8-f2229420b064)

Reply:
Good point. Falling back on empty string that will handle the logs and displays vulnerabilities accordingly